arch/tricore/src/common/tricore_csa.c

```diff
@@ -50,10 +50,18 @@ uintptr_t *tricore_alloc_csa(uintptr_t pc, uintptr_t sp,
 
   plcsa = (uintptr_t *)tricore_csa2addr(__mfcr(CPU_FCX));
 
+  /* DSYNC instruction should be executed immediately prior to the MTCR */
+
+  __dsync();
```

Comment:
should we define the common macro like DMB/DSB?

Reply:
It is reasonable to use intrinsic in arch internal implementation. similar to other architecture, tricore also has corresponding definitions:
https://github.com/anchao/tc397_sdk/blob/master/Libraries/iLLD/TC39B/Tricore/Cpu/Std/IfxCpu_IntrinsicsGcc.h#L1443-L1446

Reply:
__dsync is compiler specific intrinsic, different compiler may map the barrier to different name. macro help you migrate to the different compiler in the future.